### PR TITLE
#trhDpW7E Create roles and set permissions

### PR DESCRIPTION
**Pull request by TuyizeeAnastase**
**What does this PR do?**
Creating Roles and set permissions

**Description of task to be completed**
-creating seedings
-Admin login verification token
-creating API for creating roles and set permissions

**How can be manually tested**
-clone this repo
-run `npm install`
-run `npm dev`
-login as admin an copy token and paste it in headers x-access-token
-in postman send request on **https://eagles-phant-ft-creatin-qbwnn6.herokuapp.com/api/v1/users/:userId**

**trello story**
https://trello.com/c/trhDpW7E/9-create-roles-and-set-permissions

### CODE REVIEW

File: src/controllers/roles.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import models from '../database/models';
+
```
Comment (Ntwali-Josue) [inline src/controllers/roles.js]: remove this space

File: src/controllers/roles.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      role,
+    });
+  } catch (error) {
+    return res.status(500).json({error: error.message})
+  }
+}
+
+module.exports = {
+  createRole,
+}
```
Comment (Ntwali-Josue) [inline src/controllers/roles.js]: add an empty line at the end of this file

**Review by Ntwali-Josue — APPROVED** [review]

**Comment by Ally4** [comment]
I would like to ask you to edit the title of the PR

**Review by Mugema123 — APPROVED** [review]

**Review by Ally4 — APPROVED** [review]

**Review by blaise82 — COMMENTED** [review]
improve your description of the task to be completed

**Comment by blaise82** [comment]
Raise your coverage

**Comment by blaise82** [comment]
don't return the password while fetching all users

File: .travis.yml [hunk 3]
```diff
@@ -2,6 +2,9 @@ services:
   - postgresql
 before_script:
   - psql -c 'CREATE DATABASE test;' -U postgres
+  - psql -c 'CREATE DATABASE users;' -U postgres
+  - psql -c 'CREATE DATABASE phantom1;' -U postgres
```
Comment (blaise82) [inline .travis.yml]: make sure that only the testing database is created during testing
so you can remove these

File: package.json [hunk 4]
```diff
@@ -4,19 +4,36 @@
...
+      "src/test",
+      "src/database/models",
+      "src/database/seeders",
+      "src/controllers/userController.js",
+      "src/controllers/auth.js",
+      "src/controllers/controller.js",
+      "src/controllers/users.js",
+       "src/validations/signup.js",
+       "src/utils/mails.js",
+       "src/middleware/isAdmin.js",
```
Comment (blaise82) [inline package.json]: remove these files form exclusion

**Review by blaise82 — COMMENTED** [review]
remove all console.log() you added

File: src/controllers/auth.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
+import model from '../database/models';
+import jwt from 'jsonwebtoken';
+import { promisify } from 'util';
+import { execFileSync } from 'child_process';
+require('dotenv').config();
```
Comment (blaise82) [inline src/controllers/auth.js]: don't use require

File: src/controllers/auth.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        expiresIn:process.env.expiresIn
+    })
+};
+
+const logIn=async(req,res)=>{
+    const  { email, password }=req.body;
+    console.log(email,password);
+     if(!email || !password){
+        res.status(400).send({message:'Please enter all credentials'})
+     };
```
Comment (blaise82) [inline src/controllers/auth.js]: use validation middleware to display proper messages 
eg: Please enter email rather than Please enter all credentials

**Review by blaise82 — APPROVED** [review]

**Review by ceelogre — APPROVED** [review]

**Pull request merged**